documentation/managing_modules.md

```diff
@@ -0,0 +1,183 @@
+`apache` module, the `modules` key in your `bolt-project.yaml` file looks
+something like this:
+
+```yaml
+name: example_project
+modules:
+- name: puppetlabs-apache
+```
+
+The project's Puppetfile lists the `apache` module and all of its
+dependencies:
+
+```puppet
+# This Puppetfile is managed by Bolt. Do not edit.
+mod "puppetlabs-apache", "5.5.0"
+mod "puppetlabs-stdlib", "6.4.0"
+mod "puppetlabs-concat", "6.2.0"
+mod "puppetlabs-translate", "2.2.0"
+```
+
+And the `.modules` directory contains all of the installed modules:
+
+```shell
+$ ls modules
+apache    concat    stdlib    translate
+```
+
+## Opting in
+
+Use `bolt project migrate` to migrate a Bolt project to the new workflows. 
+
+Bolt reads your Puppetfile and prompts you to choose which modules are direct
+dependencies of your project. Only select a module as a direct dependency if
+your project uses content from that module.
+
+After you've selected your project's direct dependencies, Bolt adds those
+modules to your `bolt-project.yaml` file and pins them to their current
+versions, installs those modules and their dependencies into a `.modules`
+directory, and generates a Puppetfile. 
+
+If you don't want your modules pinned to a specific version, edit out the
+`version_requirement` fields for each module in your `bolt-project.yaml` file
+and run `bolt module install`.
+
+> **NOTE:** Migrating a project makes irreversible changes to the project's
+  configuration and inventory files. Before you migrate, make sure the project
+  has a backup or uses a version control system.
+
+## Install modules in a Bolt project
+
+Once you've opted in to the experimental workflows, you can use the `bolt module
+add` and `bolt module install` commands to manage your Forge modules. Git
+modules are not currently supported.
+
+### Add a Forge module and its dependencies to an existing project
+
+To add a single module and its dependencies to your Bolt project, use the 
+`bolt module add` command. For example, to add the puppetlabs/apt module:
+
+*\*nix shell command*
+```shell
+bolt module add puppetlabs/apt
+```
+
+*Powershell cmdlet*
+```shell
+Add-BoltModule -Module puppetlabs/apt
+```
+
+Bolt adds a declaration to your project configuration file, resolves
+dependencies, updates your project's Puppetfile, and installs the modules to the
+project's `.modules` directory.
+
+### Install the modules associated with a Bolt project
+
+If you've just downloaded a Bolt project from source control and want to install
+the modules associated with it, use the following command:
+
+*\*nix shell command*
+```shell
+bolt module install
+```
+
+*Powershell cmdlet*
+```shell
+Install-BoltModule
+```
+
+Bolt loads your `bolt-project.yaml` file, resolves dependencies for all of the
+declared modules, writes a Puppetfile to your project that includes
+specifications for the resolved modules, and installs the modules to the
+project's `.modules` directory.
+
+## How Bolt resolves shared dependencies
```

Comment:
This bit only applies to `bolt module add`. I think it gives a little more information than the user needs, so maybe something simpler under "Add a Forge module and its dependencies..." like this?

```
When Bolt adds a new module to the project and resolves its dependencies, it 
attempts to keep all installed modules on the same versions. If Bolt is unable to
do this due to a version conflict, it may update installed modules to newer versions.
```